lib/pure/times.nim

```diff
@@ -1067,6 +1134,37 @@ proc evaluateInterval(dt: DateTime, interval: TimeInterval): tuple[adjDur, absDu
     minutes = interval.minutes,
     hours = interval.hours)
 
+
+proc initDateTime*(monthday: MonthdayRange, month: Month, year: int,
+                   hour: HourRange, minute: MinuteRange, second: SecondRange,
+                   nanosecond: NanosecondRange, zone: Timezone = local()): DateTime =
+  ## Create a new ``DateTime`` in the specified timezone.
```

Comment:
Maybe to add ```milliseconds``` and ```microseconds```?

Reply:
Only having one sub second unit for `DateTime` is fairly common.

 `initDateTime(01, mJan, 2018, 00, 00, 00, convert(Milliseconds, Nanoseconds, 10))` can be used for specifying the sub second part in another unit.

Reply:
```convert(Milliseconds, Nanoseconds, 10))```
It's very cumbersome.

Reply:
Any real-life examples then it's needed? I, personally, don't think it's worth a candle

Reply:
Something like Sleep proc.

Reply:
Use it with Duration, then. I don't see the case there you'll need more than `weeks` for sleep :)

Reply:
So there are weeks and months!
I propose to add ```milliseconds``` and ```microseconds```.

Reply:
You can also use `initDateTime(...) + 10.microseconds` which is fairly short. IMO `initDateTime` is rarely used anyway, there is no need for a more complex proc signature.